lib/ansible/modules/cloud/cloudstack/cs_template.py

```diff
@@ -495,6 +507,56 @@ def create_template(self):
 
         return template
 
+    def upload_template(self):
+        required_params = [
+            'format',
+            'file',
+            'hypervisor',
+            'zone',
+        ]
+        self.module.fail_on_missing_params(required_params=required_params)
+        filepath = self.module.params.get('file')
+        if not os.path.isfile(filepath):
+            if os.path.isfile('files/' + filepath):
+                filepath = 'files/' + filepath
+            else:
+                self.module.fail_json(msg="Failed: upload file '%s' does not exist" % filepath)
+
+        template = self.get_template()
+        if not template:
+            self.result['changed'] = True
+            args                    = self._get_args()
```

Comment:
I think it would look weird if I only pep8 that one line. I'll pep8 the whole block.